### PR TITLE
Refactoring Event Calendar Component Part 1

### LINKED ISSUE #184: Refactoring Event Calendar Component Part 1
The event calendar on the main site currently has all event item components hard-coded and is nondynamic! Every time we want to add a new event, we have to add a new component.

### We want to be able to:

- [ ] Pull event data from a data file and map out each event item in the calendar. 
- [ ] If an event has not elapsed yet, the background color should be green
Screen Shot 2023-05-15 at 3 26 16 PM

- [ ] If an event has elapsed, the background color should be brown
Screen Shot 2023-05-15 at 3 14 57 PM

The end goal is to be able to add/remove events just by adding and removing event data from the data file and have it auto-update styles based on dates

### PR DESCRIPTION
**Pull request by jmsargent10**
Conditionally change box color if the event has passed; add new events dynamically

Fixes #184

Changelist:

- Updated desktop.ts and nondesktop.ts files to map out fields from objects so that new events can be added into the data file and will be displayed automatically rather than having to be hardcoded
- made it so box color is dependent on date; i.e. if the event has passed the box will be brown rather than green
- added the EventsCalendarData interface to the types.ts file and the eventsCalendarData variable in the data.ts file to hold all of the event calendar data in individual fields

Notes:

- After refactoring, we realized that desktop.ts and nondesktop.ts are very similar and could potentially be combined into one file. @mike10911 thoughts on that?
- When resizing the screen, the cactus occasionally covers the bottom right corner of the box

Tested:

- Ran main site locally to check if changes are reflected on desktop and non desktop screen widths
- Added test events that did / didn't have certain optional fields (edge cases) to check they rendered properly

Screenshots & Screencasts:
Screen Shot 2023-05-20 at 6 56 47 PM

Screen Shot 2023-05-20 at 6 57 45 PM

Screen Shot 2023-05-20 at 7 15 50 PM

Rendering with various fields pulled out:

Screen Shot 2023-05-20 at 7 17 02 PM

Screen Shot 2023-05-20 at 7 17 19 PM

Screen Shot 2023-05-20 at 7 18 09 PM

### CODE REVIEW

**Review by kzeng24 — APPROVED** [review]
Looks great!! One question I have is, what's the difference between desktop.ts and nondesktop.ts?

**Review by Aaryan1203 — COMMENTED** [review]
lgtm!

**Comment by jmsargent10** [comment]
> Looks great!! One question I have is, what's the difference between desktop.ts and nondesktop.ts?

They are currently the same, hence our questioning of if its okay to remove one of the files. The font resizing happens because of the @media in the styles file, so I think having separate files for desktop and nondesktop is redundant, at least for now!

File: main-site/components/events-calendar-section/texts/DesktopTexts.tsx [hunk 1]
```diff
@@ -1,78 +1,46 @@
...
   StyledDesktopTextsContainer,
-  // StyledCybersecurityLocationContainer
+  StyledCybersecurityLocationContainer
 } from '../EventsCalendarSection.styles';
+import { EventsCalendarData } from '../../../lib/types';
+import { eventsCalendarData } from '../../../lib/data';
 
 const DesktopTexts: React.FC = () => {
+  const events: EventsCalendarData[] = eventsCalendarData;
+  let CurrentDate = new Date();
```
Comment (mike10911) [inline main-site/components/events-calendar-section/texts/DesktopTexts.tsx]: Variables should be defined in camel case
**Ex:**
`camelCase`

Comment (mike10911) [inline main-site/components/events-calendar-section/texts/DesktopTexts.tsx]: The current date can also be defined with `const` since it is never mutated. We want to use `let` if the variable is going to change between component rerenders. In this case, we only need the date each time the component rerenders.

File: main-site/components/events-calendar-section/texts/DesktopTexts.tsx [hunk 2]
```diff
@@ -1,78 +1,46 @@
...
+  TextLeftContainer,
   StyledDesktopTextsContainer,
-  // StyledCybersecurityLocationContainer
+  StyledCybersecurityLocationContainer
 } from '../EventsCalendarSection.styles';
+import { EventsCalendarData } from '../../../lib/types';
+import { eventsCalendarData } from '../../../lib/data';
 
 const DesktopTexts: React.FC = () => {
+  const events: EventsCalendarData[] = eventsCalendarData;
```
Comment (mike10911) [inline main-site/components/events-calendar-section/texts/DesktopTexts.tsx]: We don't have to redefine the list of event objects as it is already defined and typed in `data.ts`. You can simply use the import like you would a variable.

File: main-site/components/events-calendar-section/texts/DesktopTexts.tsx [hunk 3]
```diff
@@ -1,78 +1,46 @@
...
+  EventsP,
   StyledEventsBox,
   StyledTextContainer,
-  // EventsLocationP,
-  // TextLeftContainer,
+  EventsLocationP,
+  TextLeftContainer,
   StyledDesktopTextsContainer,
-  // StyledCybersecurityLocationContainer
+  StyledCybersecurityLocationContainer
```
Comment (mike10911) [inline main-site/components/events-calendar-section/texts/DesktopTexts.tsx]: We can delete this import and the component itself

File: main-site/components/events-calendar-section/EventsCalendarSection.styles.ts [hunk 4]
```diff
@@ -23,10 +24,10 @@ const StyledEventsContainer = styled.div`
   justify-content: center;
 `;
 
-const StyledEventsBox = styled.div`
+const StyledEventsBox = styled.div<{ elapsedEvent: boolean}>`
   border-radius: 2em;
   
-  background-color: ${colors.BUTTON_DARK_GREEN};
+  background-color: ${(props) => (props.elapsedEvent ? colors.TEXT_BROWN : colors.BUTTON_DARK_GREEN)};
```
Comment (mike10911) [inline main-site/components/events-calendar-section/EventsCalendarSection.styles.ts]: Because this is a function, typescript requires you to type it!

**Review by mike10911 — CHANGES_REQUESTED** [review]
Great job with the refractor!!! This is a core part of react methodologies so it is great you are getting your hands dirty with it! Here are some needed changes

- To answer @kzeng24 and @jmsargent10—isDesktop and isMobile are boolean variables that are set to determine the viewport size. Once the viewport becomes the respective sizes, the boolean asserts as true. We use these variables here to conditionally render the mobile and desktop versions of the event components. This is certainly a valid way of rendering responsive components it is usually implemented when the mobile and desktop versions of a component are drastically different. In this case, the two versions of the components are similar enough to have 1 component with both versions and utilize media queries—and other CSS properties—to ensure responsive design.
- I’m seeing that some commits from your ticket last week are in the commit history on this branch. This is most likely the result of development on your old branch. For next time, be sure to create a fresh branch from `main`!
- In the data.ts file, I see that you added a new interface for the events. Be sure to add this to the change list in your PR:)

**Comment by mike10911** [comment]
@jmsargent10 Thanks for pointing out the cactus issue! That will be taken care of in a future ticket since it is not in the scope of this ticket!

**Review by mike10911 — APPROVED** [review]
Let's just change all events to have accurate years so they are all elapsed. After that LGTM!!!!
